### PR TITLE
Factor out get_signature and add it to canister_sig_util

### PR DESCRIPTION
**Pull request by przydatek**
Factor out `get_signature`, which is used in canisters that create canister signatures, and add the corresponding functionality to `canister_sig_util`-crate (as function `get_signature_as_cbor`).

<hr/><details><summary>🟡 Some screens were changed</summary></details>

### CODE REVIEW

File: src/canister_sig_util/src/lib.rs [hunk 1]
```diff
@@ -125,6 +131,63 @@ pub fn extract_raw_canister_sig_pk_from_der(pk_der: &[u8]) -> Result<Vec<u8>, St
...
+) -> Result<Vec<u8>, String> {
+    let certificate = data_certificate()
+        .ok_or("data certificate is only available in query calls".to_string())?;
+    let witness = sig_map
+        .witness(hash_bytes(seed), message_hash)
+        .ok_or("missing witness".to_string())?;
+
+    let witness_hash = witness.digest();
+    let root_hash = sig_map.root_hash();
+    if witness_hash != root_hash {
```
Comment (frederikrothenberger) [inline src/canister_sig_util/src/lib.rs:166]: Does this check (still) make sense? We rehash the pruned witness tree every time, but if did not match the (unpruned) signature map root hash we would simply have a bug in our implementation. I wonder if this is really best addressed by wasting resources in production, or if we should rather have better tests (unless we're already sufficiently covered).
Honestly, I think we should remove it here.

File: src/canister_sig_util/src/lib.rs [hunk 2]
```diff
@@ -125,6 +131,63 @@ pub fn extract_raw_canister_sig_pk_from_der(pk_der: &[u8]) -> Result<Vec<u8>, St
...
+        return Err(format!(
+            "internal error: signature map computed an invalid hash tree, witness hash is {}, root hash is {}",
+            hex::encode(witness_hash),
```
Comment (frederikrothenberger) [inline src/canister_sig_util/src/lib.rs:169]: You could also get rid of the `hex` dependency if you didn't have this check 😉

**Review by frederikrothenberger — APPROVED** [review]
LGTM, thanks!

File: src/canister_sig_util/src/lib.rs (hunk 1 above)
Comment (przydatek) [reply src/canister_sig_util/src/lib.rs:166]: Filed https://dfinity.atlassian.net/browse/GIX-2178

**Review by przydatek — COMMENTED** [review]
Thanks, I've filed a ticket for the potential code simplification.

**Pull request merged**